### PR TITLE
chore(CDAP-20449): Move the branch create step in Before and delete step in After

### PR DESCRIPTION
**Pull request by de-lan**
Move the branch create step in Before and delete step in After

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [x] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-20449](https://cdap.atlassian.net/browse/CDAP-20449)

[CDAP-20449]: https://cdap.atlassian.net/browse/CDAP-20449?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by sumengwang — APPROVED** [review]

**Review by samdgupi — APPROVED** [review]
LGTM

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/BeforeActions.java [hunk 1]
```diff
@@ -36,13 +37,23 @@ public void loginIfRequired() throws IOException {
...
-  public void setGitBranchConfig() {
+  @Before(order = 1, value = "@SOURCE_CONTROL_MANAGEMENT_CONFIG_CRUD_TEST")
+  public void setGitRepoConfig() {
+    addRepoCredentials();
+  }
+
+  @Before(order = 1, value = "@SOURCE_CONTROL_MANAGEMENT_SYNC_APPS_TEST")
+  public void setGitBranchConfig() throws GitAPIException, IOException {
+    addRepoCredentials();
     if (Strings.isNullOrEmpty(PluginPropertyUtils.pluginProp(Constants.GIT_BRANCH_PROP_NAME))) {
```
Comment (samdgupi) [inline src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/BeforeActions.java]: Is `GIT_BRANCH_PROP_NAME ` not needed in SOURCE_CONTROL_MANAGEMENT_CONFIG_CRUD_TEST also ?
We can make `setGitBranchConfig` -> `createGitBranchConfig`

File: src/e2e-test/java/io/cdap/cdap/ui/utils/Constants.java [hunk 2]
```diff
@@ -89,9 +89,10 @@ private static double getRandomArbitrary(int min, int max) {
   public static final String FAKE_REPO_LINK = "https://example.com";
   public static final String FAKE_TOKEN_NAME = "fake_token";
   public static final String FAKE_TOKEN = "fake token value";
-  public static final String FAKE_DEFAULT_BRANCH = "noneExistingBranch";
+  public static final String NON_EXITS_DEFAULT_BRANCH = "noneExistingBranch";
```
Comment (samdgupi) [inline src/e2e-test/java/io/cdap/cdap/ui/utils/Constants.java]: EXISTS

**Review by arjan-bal — APPROVED** [review]

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/BeforeActions.java (hunk 1 above)
Comment (de-lan) [reply src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/BeforeActions.java]: hmmm, good point, actually we can add the test step for the branch as well

Comment (de-lan) [reply src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/BeforeActions.java]: Then I don't need to have two suite test tags. Just have one `SOURCE_CONTROL_MANAGEMENT_TEST` and create and delete the branch in all SCM related tests.

**Pull request merged**